examples/esp32/xively_demo/main/main.c

```diff
@@ -422,7 +422,26 @@ void esp32_xibsp_notify_chunk_written( size_t chunk_size, size_t offset )
     }
 }
 
-void esp32_xibsp_notify_update_applied()
+void esp32_xibsp_notify_update_applied( uint8_t app_updated )
```

Comment:
This is application-level logic (rebooting the device or not based on the package contents), so I've taken this opportunity to move it out of the BSP.